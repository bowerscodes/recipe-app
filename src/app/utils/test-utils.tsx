import { ReactNode } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { HeroUIProvider } from "@heroui/react";
import RecipeProvider from "@/context/RecipeContext";

const AllProviders = ({ children }: { children: ReactNode }) => (
  <HeroUIProvider>
    <RecipeProvider>{children}</RecipeProvider>
  </HeroUIProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
