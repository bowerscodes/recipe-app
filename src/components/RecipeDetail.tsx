import { Card, CardBody, CardHeader } from "@heroui/card";
import { Link } from "@heroui/link";

import { Recipe } from "@/types/Recipe";

interface RecipeDetailProps {
  recipe: Recipe;
}

export default function RecipeDetail({ recipe }: RecipeDetailProps) {
  return (
    <div className="mt-6 mx-10">
      <Link href="/">‚Üê Back to Recipes</Link>

      <div className="mt-2">
        <div
          className="relative bg-cover bg-center bg-no-repeat rounded-lg p-6 h-[200px] flex flex-col justify-end"
          style={{ backgroundImage: `url(${recipe.image})` }}
        >

          {/* Header Image tint */}
          <div
            className="absolute inset-0 rounded-lg"
            style={{ backgroundColor: "rgba(0, 0, 0, 0.2)" }}
          />
          
          <h1 className="text-white mb-0 z-10">{recipe.title}</h1>
          <p className="text-white font-semibold z-10">{recipe.cookingTime}</p>
        </div>

        <div className="mt-4">
          <Card>
            <CardHeader>
              <h2>Ingredients</h2>
            </CardHeader>
            <CardBody>
              <ul>
                {recipe.ingredients.map((ingredient, index) => (
                  <li key={index}>
                    <span>{ingredient}</span>
                  </li>
                ))}
              </ul>
            </CardBody>
          </Card>
        </div>

        <div className="mt-4">
          <Card>
            <CardHeader>
              <h2>Instructions</h2>
            </CardHeader>
            <CardBody>
              <ol>
                {recipe.instructions.map((instruction, index) => (
                  <li
                    key={index}
                    className="grid grid-cols-[2rem_1fr] gap-2 mb-2"
                  >
                    <span className="text-right">{index + 1 + "."}</span>
                    <span>{instruction}</span>
                  </li>
                ))}
              </ol>
            </CardBody>
          </Card>
        </div>

        <div className="mt-4">
          <Card>
            <CardHeader>
              <h3>Reviews</h3>
            </CardHeader>
            <ul className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 px-3 pb-3">
              {recipe.reviews.map((review, index) => (
                <Card key={index}>
                  <li className="bg-white p-4 flex flex-col items-center">
                    <span className="font-bold">{review.rating + " / 5 "}</span>
                    {<em>&quot;{review.comment}&quot;</em>}
                  </li>
                </Card>
              ))}
            </ul>
          </Card>
        </div>
      </div>
    </div>
  );
}
