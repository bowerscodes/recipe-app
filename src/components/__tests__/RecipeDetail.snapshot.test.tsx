import { render } from "@/app/utils/test-utils";
import RecipeDetail from "../RecipeDetail";
import recipesData from "@/data/recipes.json";

const recipe = recipesData[0];

describe("RecipeDetail Snapshots", () => {
  test("renders RecipeDetail correctly", () => {
    const { container } = render(<RecipeDetail recipe={recipe} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders RecipeDetail with user-created recipe", () => {
    const userRecipe = {
      id: 1001,
      title: "My Custom Recipe",
      image: "https://example.com/custom.jpg",
      cookingTime: "30 minutes",
      ingredients: [
        "2 cups flour",
        "1 cup sugar",
        "3 eggs"
      ],
      instructions: [
        "Mix all dry ingredients",
        "Add wet ingredients",
        "Bake for 30 minutes"
      ],
      reviews: [
        { rating: 5, comment: "Amazing recipe!" }
      ],
      author: "me"
    };

    const { container } = render(<RecipeDetail recipe={userRecipe} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders RecipeDetail with no reviews", () => {
    const recipeWithoutReviews = {
      ...recipe,
      reviews: []
    };

    const { container } = render(<RecipeDetail recipe={recipeWithoutReviews} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
