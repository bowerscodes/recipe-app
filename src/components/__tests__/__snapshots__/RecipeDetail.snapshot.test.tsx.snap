// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RecipeDetail Snapshots renders RecipeDetail correctly 1`] = `
<div>
  <div
    class="mt-6 mx-10"
  >
    <a
      class="relative inline-flex items-center tap-highlight-transparent outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-hover active:opacity-disabled transition-opacity"
      data-react-aria-pressable="true"
      href="/"
      role="link"
      tabindex="0"
    >
      ‚Üê Back to Recipes
    </a>
    <div
      class="mt-2"
    >
      <div
        class="relative bg-cover bg-center bg-no-repeat rounded-lg p-6 h-[200px] flex flex-col justify-end"
        style="background-image: url("https://img2.thejournal.ie/article/5084219/river/?height=400&version=5084251");"
      >
        <div
          class="absolute inset-0 rounded-lg"
          style="background-color: rgba(0, 0, 0, 0.2);"
        />
        <h1
          class="text-white mb-0 z-10"
        >
          Lasagne & Chips
        </h1>
        <p
          class="text-white font-semibold z-10"
        >
          45 minutes
        </p>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h2>
              Ingredients
            </h2>
          </div>
          <div
            class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
          >
            <ul>
              <li>
                <span>
                  200g Lasagne Sheets
                </span>
              </li>
              <li>
                <span>
                  500g Minced Beef
                </span>
              </li>
              <li>
                <span>
                  1 onion
                </span>
              </li>
              <li>
                <span>
                  4 cloves Garlic
                </span>
              </li>
              <li>
                <span>
                  Tomato Sauce
                </span>
              </li>
              <li>
                <span>
                  Cheese Sauce
                </span>
              </li>
              <li>
                <span>
                  Oven Chips
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h2>
              Instructions
            </h2>
          </div>
          <div
            class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
          >
            <ol>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  1.
                </span>
                <span>
                  Fry the onions and garlic in a medium frying pan until softened
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  2.
                </span>
                <span>
                  Add the mince to the pan and brown evenly
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  3.
                </span>
                <span>
                  Once browned, stir in the tomato sauce and simmer for 10 minutes
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  4.
                </span>
                <span>
                  Pre-heat oven to 180 degrees Celcius
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  5.
                </span>
                <span>
                  Line a Baking Dish with Lasagne Sheets and spoon in half of the beef mixture
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  6.
                </span>
                <span>
                  Add another layer of lasagne sheets on top, and add half of the jar of cheese sauce
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  7.
                </span>
                <span>
                  Repeat by adding the remaining beef mixture, topped with another layer of lasagne sheets, followed by Cheese sauce
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  8.
                </span>
                <span>
                  Add the lasagne and Chips to the oven and bake for 25 minutes.
                </span>
              </li>
            </ol>
          </div>
        </div>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h3>
              Reviews
            </h3>
          </div>
          <ul
            class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 px-3 pb-3"
          >
            <div
              class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
              tabindex="-1"
            >
              <li
                class="bg-white p-4 flex flex-col items-center"
              >
                <span
                  class="font-bold"
                >
                  5 / 5 
                </span>
                <em>
                  "
                  Delicious! Best lasagne I've ever had
                  "
                </em>
              </li>
            </div>
            <div
              class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
              tabindex="-1"
            >
              <li
                class="bg-white p-4 flex flex-col items-center"
              >
                <span
                  class="font-bold"
                >
                  4 / 5 
                </span>
                <em>
                  "
                  Tasty, but could do with some grated cheese on top
                  "
                </em>
              </li>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RecipeDetail Snapshots renders RecipeDetail with no reviews 1`] = `
<div>
  <div
    class="mt-6 mx-10"
  >
    <a
      class="relative inline-flex items-center tap-highlight-transparent outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-hover active:opacity-disabled transition-opacity"
      data-react-aria-pressable="true"
      href="/"
      role="link"
      tabindex="0"
    >
      ‚Üê Back to Recipes
    </a>
    <div
      class="mt-2"
    >
      <div
        class="relative bg-cover bg-center bg-no-repeat rounded-lg p-6 h-[200px] flex flex-col justify-end"
        style="background-image: url("https://img2.thejournal.ie/article/5084219/river/?height=400&version=5084251");"
      >
        <div
          class="absolute inset-0 rounded-lg"
          style="background-color: rgba(0, 0, 0, 0.2);"
        />
        <h1
          class="text-white mb-0 z-10"
        >
          Lasagne & Chips
        </h1>
        <p
          class="text-white font-semibold z-10"
        >
          45 minutes
        </p>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h2>
              Ingredients
            </h2>
          </div>
          <div
            class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
          >
            <ul>
              <li>
                <span>
                  200g Lasagne Sheets
                </span>
              </li>
              <li>
                <span>
                  500g Minced Beef
                </span>
              </li>
              <li>
                <span>
                  1 onion
                </span>
              </li>
              <li>
                <span>
                  4 cloves Garlic
                </span>
              </li>
              <li>
                <span>
                  Tomato Sauce
                </span>
              </li>
              <li>
                <span>
                  Cheese Sauce
                </span>
              </li>
              <li>
                <span>
                  Oven Chips
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h2>
              Instructions
            </h2>
          </div>
          <div
            class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
          >
            <ol>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  1.
                </span>
                <span>
                  Fry the onions and garlic in a medium frying pan until softened
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  2.
                </span>
                <span>
                  Add the mince to the pan and brown evenly
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  3.
                </span>
                <span>
                  Once browned, stir in the tomato sauce and simmer for 10 minutes
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  4.
                </span>
                <span>
                  Pre-heat oven to 180 degrees Celcius
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  5.
                </span>
                <span>
                  Line a Baking Dish with Lasagne Sheets and spoon in half of the beef mixture
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  6.
                </span>
                <span>
                  Add another layer of lasagne sheets on top, and add half of the jar of cheese sauce
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  7.
                </span>
                <span>
                  Repeat by adding the remaining beef mixture, topped with another layer of lasagne sheets, followed by Cheese sauce
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  8.
                </span>
                <span>
                  Add the lasagne and Chips to the oven and bake for 25 minutes.
                </span>
              </li>
            </ol>
          </div>
        </div>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h3>
              Reviews
            </h3>
          </div>
          <ul
            class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 px-3 pb-3"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RecipeDetail Snapshots renders RecipeDetail with user-created recipe 1`] = `
<div>
  <div
    class="mt-6 mx-10"
  >
    <a
      class="relative inline-flex items-center tap-highlight-transparent outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-hover active:opacity-disabled transition-opacity"
      data-react-aria-pressable="true"
      href="/"
      role="link"
      tabindex="0"
    >
      ‚Üê Back to Recipes
    </a>
    <div
      class="mt-2"
    >
      <div
        class="relative bg-cover bg-center bg-no-repeat rounded-lg p-6 h-[200px] flex flex-col justify-end"
        style="background-image: url("https://example.com/custom.jpg");"
      >
        <div
          class="absolute inset-0 rounded-lg"
          style="background-color: rgba(0, 0, 0, 0.2);"
        />
        <h1
          class="text-white mb-0 z-10"
        >
          My Custom Recipe
        </h1>
        <p
          class="text-white font-semibold z-10"
        >
          30 minutes
        </p>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h2>
              Ingredients
            </h2>
          </div>
          <div
            class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
          >
            <ul>
              <li>
                <span>
                  2 cups flour
                </span>
              </li>
              <li>
                <span>
                  1 cup sugar
                </span>
              </li>
              <li>
                <span>
                  3 eggs
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h2>
              Instructions
            </h2>
          </div>
          <div
            class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
          >
            <ol>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  1.
                </span>
                <span>
                  Mix all dry ingredients
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  2.
                </span>
                <span>
                  Add wet ingredients
                </span>
              </li>
              <li
                class="grid grid-cols-[2rem_1fr] gap-2 mb-2"
              >
                <span
                  class="text-right"
                >
                  3.
                </span>
                <span>
                  Bake for 30 minutes
                </span>
              </li>
            </ol>
          </div>
        </div>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
          tabindex="-1"
        >
          <div
            class="flex p-3 z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large"
          >
            <h3>
              Reviews
            </h3>
          </div>
          <ul
            class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 px-3 pb-3"
          >
            <div
              class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none"
              tabindex="-1"
            >
              <li
                class="bg-white p-4 flex flex-col items-center"
              >
                <span
                  class="font-bold"
                >
                  5 / 5 
                </span>
                <em>
                  "
                  Amazing recipe!
                  "
                </em>
              </li>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
